python/ray/data/datasource/file_based_datasource.py

```diff
@@ -65,31 +67,70 @@ class FileShuffleConfig:
     """Configuration for file shuffling.
 
     This configuration object controls how files are shuffled while reading file-based
-    datasets.
+    datasets. You can configure either a fixed seed for consistent shuffling across
+    epochs, or a base seed for different shuffling in each epoch.
 
     .. note::
-        Even if you provided a seed, you might still observe a non-deterministic row
-        order. This is because tasks are executed in parallel and their completion
-        order might vary. If you need to preserve the order of rows, set
-        `DataContext.get_current().execution_options.preserve_order`.
+        Even if you provided a seed or base_seed, you might still observe a
+        non-deterministic row order. This is because tasks are executed in parallel
+        and their completion order might vary. If you need to preserve the order of
+        rows, set ``DataContext.get_current().execution_options.preserve_order``.
 
     Args:
         seed: An optional integer seed for the file shuffler. If provided, Ray Data
-            shuffles files deterministically based on this seed.
+            shuffles files deterministically with the same order across all epochs.
+            Cannot be set together with ``base_seed``.
+        base_seed: An optional integer base seed for the file shuffler. If provided,
+            Ray Data shuffles files deterministically but with different orders across
+            epochs. The actual seed used for each epoch is computed as
+            ``base_seed + epoch_idx``. Cannot be set together with ``seed``.
 
     Example:
         >>> import ray
         >>> from ray.data import FileShuffleConfig
+        >>> # Fixed seed - same shuffle across epochs
         >>> shuffle = FileShuffleConfig(seed=42)
         >>> ds = ray.data.read_images("s3://anonymous@ray-example-data/batoidea", shuffle=shuffle)
+        >>>
+        >>> # Base seed - different shuffle per epoch
+        >>> shuffle = FileShuffleConfig(base_seed=42)
+        >>> ds = ray.data.read_images("s3://anonymous@ray-example-data/batoidea", shuffle=shuffle)
     """  # noqa: E501
 
     seed: Optional[int] = None
+    base_seed: Optional[int] = None
```

Comment:
it'd be easier to understand if we expose a `reseed_after_epoch` bool flag in the user API. the base_seed can be an internal implementation.